lib/srv/usermgmt.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package srv
+
+import (
+	"os/user"
+	"runtime"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/siddontang/go/log"
+)
+
+func NewUserManagement() (UserManagement, error) {
+	if runtime.GOOS == "windows" {
+		return nil, trace.NotImplemented("Host user creation management is only supported on linux")
+	}
+	return &unixMgmt{}, nil
+}
+
+type UserManagement interface {
```

Comment:
Nit: Let's call it `HostUsersBackend` - it will plug into your `HostUsers` interface implementation that I mention below. Also, add Godocs on the interface and its public methods.